### ISSUE TITLE
core/social-link component doesn't render accessible markup

### ISSUE THREAD

**Issue by 0aveRyan** [WordPress/gutenberg#18620]
**Describe the bug**
The anchor markup rendered by `core/social-link` isn't accessible. Anchors need [text to describe the links](https://dequeuniversity.com/rules/axe/3.1/link-name?application=lighthouse) to a screen reader.

[Render method](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/social-link/index.php#L25)

**To reproduce**
1. Go to Block Editor
2. Insert the **Social Links** block
3. Click **Preview** to see rendered block markup
4. Inspect Anchors or run Google Lighthouse, which will flag the issue as `Links do not have a discernible name`, multiplied by the number of inserted anchors.

**Expected behavior**
In the current implementation, anchor elements simply wrap SVGs of each social brand. The name of each site is already available and used to add a unique CSS class, so the fix should be to a fairly light lift to add a default, descriptive `aria-label="{BRAND} logo"` to the `<a>`.

However, we might also consider adding a custom override for the AIRA label in `core/social-link`. If a blog post was created identifying social profiles for **_10 Fast-Food Brands That Have Fire Social Accounts_**, it's possible there could be many instances of the Social Links block. Having a screen reader read "Twitter logo" 10+ times throughout the document isn't nearly as useful as... 

- "Twitter list for 10 Fast-Food Brands That Have Fire Social Accounts"
- "Twitter profile for Shake Shack", 
- "Twitter profile for Popeyes" 
- and "Twitter profile for AwesomeFastFoodReviews.com" [the current site].

**Comment by 0aveRyan** [comment]
@gziolo I've opened a PR to address this in #18651
